soh/soh/Enhancements/randomizer/randomizer_entrance_tracker.cpp

```diff
@@ -886,17 +901,50 @@ void EntranceTrackerWindow::DrawElement() {
                     // Use a non-breaking space to keep the arrow from wrapping to a newline by itself
                     auto nbsp = u8"\u00A0";
                     if (original->srcGroup != ENTRANCE_GROUP_ONE_WAY) {
-                        ImGui::TextWrapped("%s to %s%s->", origSrcName, origDstName, nbsp);
+                        if (!destToggle) ImGui::PushStyleColor(ImGuiCol_Text, COLOR_GRAY);  
+                        if (locationSearch.IsActive() || destToggle || !CVarGetInteger("gTrackers.EntranceTracker.SimplifyTo", 0)) {
+                            ImGui::Text("%s", origSrcName);
+                            ImGui::SameLine();
+                        }
+                        if (!destToggle) ImGui::PopStyleColor();
+                        
+                        if (destToggle) ImGui::PushStyleColor(ImGuiCol_Text, COLOR_GRAY);
+                        if (locationSearch.IsActive() || !destToggle || !CVarGetInteger("gTrackers.EntranceTracker.SimplifyTo", 0))
+                            ImGui::Text("%s%s%s->", CVarGetInteger("gTrackers.EntranceTracker.SimplifyTo", 0) && !locationSearch.IsActive() ? "" : "to ", origDstName, nbsp);
+                        else
+                            ImGui::Text("->");
+                        if (destToggle) ImGui::PopStyleColor();
```

Comment:
I think this would be cleaner if it used braces for each of the ifs, especially for the if(locationSearch.... since its not clear at first glance what is and isnt part of the if statement

Reply:
the same is true for later on too